lib/Gateway.js

```diff
@@ -1265,17 +1265,21 @@ Gateway.prototype.discoverValue = function (node, vId) {
       case CommandClasses['Binary Sensor']:
         // https://github.com/zwave-js/node-zwave-js/blob/master/packages/zwave-js/src/lib/commandclass/BinarySensorCC.ts#L41
 
-        let sensorTypeName = BinarySensorType[valueId.property]
+        // change the sensorTypeName to use directly valueId.property, as the old way was returning a number
+        // add a comment which shows the old way of achieving this value. This change fixes the Binary Sensor
+        // discovery
+        let sensorTypeName = valueId.property
```

Comment:
I would try to use the `ccSpecific.sensorType` metadata property (numeric enum value) if it exists and only fall back to the `property`.

Reply:
My understanding this comes after metadata. and is parsed as valueId.

Reply:
My understanding is, valueId is internal of zwavejs2mqtt. and is already translated during the device discovery. this is homeassistant sensor discovery. afterwards

Reply:
@robertsLando ?

Reply:
@billiaz What does this fixes? I mean you say the old way was returning a number, sholdn't it return a string based on https://github.com/zwave-js/node-zwave-js/blob/master/packages/zwave-js/src/lib/commandclass/BinarySensorCC.ts#L41 enum? At least that is what I want

Reply:
@robertsLando  I have a Motion sensor. this sensor is 0x0c (12 dec) and name is Motion
I set a debug output and tried to find out what the name of sensor was.
This snippet:
```
let sensorTypeName = BinarySensorType[valueId.property]
```
Returns '0x0c' (12)

when printing on my debug output the valueid.property, I do find the string (motion, contact etc)
basically translating the above is: 
```
BinarySensorType['Motion']
```
I think using directly the string is good. 
If we want to cover both cases where we receive a number or string on valueId.property , I could create an isNaN check.
But My understanding reading code, valueId.Property is already parsed and made string

Reply:
`property` contains the **name** of the enum member. `ccSpecific.sensorType` is numeric

Reply:
Ok so this is correct :+1: